### ISSUE TITLE
support for "workflow_run" event triggers

### ISSUE THREAD

**Issue by jkowalleck** [actions/download-artifact#60]
the support for the newly introduced event "[worgkflow_run]" for github actions seams broken. 

see my test environment: https://github.com/jkowalleck/playground_workflow_run
----

# playground_workflow_run

playground for github event "[worgkflow_run]".

> For example, if your `pull_request` workflow generates build artifacts,
> you can create a new workflow that uses `workflow_run` to analyze the
> results and add a comment to the original pull request.

The goal of tis playground is to see if downloading and processing
artifacts in a "[worgkflow_run]" eent works as expected.

## expectation

in a "[worgkflow_run]" triggered event
i can download artifacts from the workflow that triggered the "[worgkflow_run]" trigger.

## conclusion

the action "[download-artifact]" semas to be broken on event "[worgkflow_run]".

## obeservations

* upload artifact in a "push"-event workflow [works as expected][action_results_tests_run].
* download artifact in the same "push"-event workflow [works as expected][action_results_tests_run].
* download artifact in a "worgkflow_run"-event workflow [is broken][action_results_tests_done].
* see [action results][action_results].

## how i tested

1. action "[TESTS][action_tests_run]" is triggered on push.
   * it creates an artifact "reports".
   * it downloads that exact sams artifact "reports".
1. action "[TESTS DONE][action_tests_done]" is triggered
   after the action "[TESTS][action_tests_run]" completed.
   * it downloads that exact sams artifact "reports" from the "TESTS" action.

[worgkflow_run]: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_run
[upload-artifact]: https://github.com/actions/upload-artifact
[download-artifact]: https://github.com/actions/download-artifact
[action_tests_done]: .github/workflows/tests_done.yaml
[action_tests_run]: .github/workflows/tests_run.yaml
[action_results]: https://github.com/jkowalleck/playground_workflow_run/actions
[action_results_tests_run]: https://github.com/jkowalleck/playground_workflow_run/actions?query=workflow%3A%22TESTS+RUN%22
[action_results_tests_done]: https://github.com/jkowalleck/playground_workflow_run/actions?query=workflow%3A%22TESTS+DONE%22

**Comment by pllim** [comment]
I think I just encountered this. I have a `pull_request` workflow that uploads the artifact, and as recommended by https://securitylab.github.com/research/github-actions-preventing-pwn-requests , I am using `workflow_run` to download it, but I encountered this error:

```
Error: Unable to find any artifacts for the associated workflow
```

Luckily the article has code to download using `github-script`, though it would have been so elegant to just use this action. I'll be interested to revisit this action when this is supported. Thank you!

**Comment by ThiefMaster** [comment]
Yeah, this not working properly is a major WTF. I mean come on, having to paste non-trivial code into my revision (from that blog post) is ridiculous. Especially since I'll have to repeat that in every repo...

**Comment by laeubi** [comment]
Any progress on this?

**Comment by t3chguy** [comment]
https://github.com/actions/download-artifact#download-artifacts-from-other-workflow-runs-or-repositories does most of this, just doesn't automagically pass through the ¦run-id`

**Comment by polarathene** [comment]
> just doesn't automagically pass through the `run-id`

Isn't that available?:

```yml
run_id: ${{ github.event.workflow_run.id }}
```

I haven't verified, but adapted the workflow from `dawidd6/action-download-artifact@v3` to `actions/download-artifact@v4`](https://github.com/docker-mailserver/docker-mailserver/pull/3838) and expect that to work fine.

**Comment by t3chguy** [comment]
Yes. That indeed has worked for me @polarathene

**Comment by polarathene** [comment]
> One unexpected catch was needing to pass a github token with additional permissions, which is unexpected

Oh? In `dawidd6/action-download-artifact@v3` I had provided the token as well as I think that was required for what I was doing 🤔 (_PR branch workflow triggers a master branch workflow_)

I did see the [docs for `actions/download-artifact@v4`](https://github.com/actions/download-artifact?tab=readme-ov-file#download-artifacts-from-other-workflow-runs-or-repositories) state a workflow_run reference will require a token with `access:read` permissions, which in my case (same repo, just different branches) that should be fine.

**Comment by t3chguy** [comment]
> (PR branch workflow triggers a master branch workflow)

This is what I am using it for too, but I did not pass any token to `action-download-artifact` e.g. https://github.com/matrix-org/matrix-react-sdk/commit/1dbcdfb6b29036798be348dc05fccda90121e66a yet ultimately had to specify a PAT to retain the same function using the 1st party action - https://github.com/matrix-org/matrix-react-sdk/commit/0d2253e7d04ba173e7d8d05b299404af93e92368
